docs/examples/uds_server_auxiliary.rst

```diff
@@ -0,0 +1,352 @@
+    import pykiso
+    from pykiso.auxiliaries import uds_aux
+
+    # only used for type-hinting the custom callback
+    from pykiso.lib.auxiliaries.udsaux import UdsServerAuxiliary
+
+    def custom_callback(ecu_reset_request: list, aux: UdsServerAuxiliary) -> None:
+        """Custom callback example for an ECU reset request.
+
+        This simulates a pending response from the server before sending the
+        corresponding positive response.
+
+        :param ecu_reset_request: received ECU reset request from the client.
+        :param aux: current UdsServerAuxiliary instance used in test.
+        """
+        for _ in range(4):
+            aux.send_response([0x7F, 0x78])
+            time.sleep(0.1)
+        aux.send_response([0x51, 0x01])
+
+
+    @pykiso.define_test_parameters(suite_id=1, case_id=1, aux_list=[uds_aux])
+    class ExampleUdsServerTest(pykiso.BasicTest):
+
+        def setUp(self):
+            """Register various callbacks for the test."""
+            # handle extended diagnostics session request
+            # respond to an incoming request [0x10, 0x03] with [0x50, 0x03, 0x12, 0x34]
+            uds_aux.register_callback(request=0x1003, response=0x50031234)
+
+            # handle incoming read data by identifier request with identifier [0x01, 0x02]
+            # the response will be built by:
+            # - creating the positive response corresponding to the request: 0x620102
+            # - appending the passed response data b'DATA': 0x620102_44415451
+            # - zero-padding the response data until the expected length is reached: 0x620102_44415451_0000
+            uds_aux.register_callback(request=0x220102, response_data=b'DATA', data_length=6)
+
+        def test_run(self):
+            """Actual test."""
+            ...
+
+        def tearDown(self):
+            """Unregister all callbacks registered by the auxiliary."""
+
+            for callback in uds_aux.callbacks:
+                uds_aux.register_callback(callback)
+
+Accessing UDS callbacks
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Once registered, callbacks can be accessed inside a test via the
+:py:attr:`~pykiso.lib.auxiliaries.udsaux.uds_server_auxiliary.UdsServerAuxiliary.callbacks`_ attribute.
+This attribute is a dictionary linking the registered request as an **uppercase** hexadecimal string
+(e.g. `"0x2E0102"`_) to the corresponding registered callback.
+
+Accessing a callback can be useful for verifying if a callback was called at some point. Based on
+the test snippets above, the following complete test example aims to show this feature and provided
+an overview of all previously described features:
+
+.. code:: python
+
+    import pykiso
+    from pykiso.auxiliaries import uds_aux
+
+    # only used for type-hinting the custom callback
+    from pykiso.lib.auxiliaries.udsaux import UdsServerAuxiliary
+
+    from uds_callback_definition import UDS_CALLBACKS
+
+    def custom_callback(ecu_reset_request: list, aux: UdsServerAuxiliary) -> None:
```

Comment:
where does it get reigistered ? 
Can you give an example for ecu_reset_request ?